core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -847,24 +847,38 @@ public String toString() {
      * This method creates a Config map out of the PMap. Later on this conversion should not be
      * necessary when we read JSON.
      */
-    public ConfigMap readStringMap(PMap weightingMap) {
-        Map<String, Double> map = new HashMap<>();
+    public Result createResult(PMap pMap) {
+        HashMap<String, Double> map = new HashMap<>(DEFAULT_SPEEDS.size());
         for (Entry<String, Double> e : DEFAULT_SPEEDS.entrySet()) {
-            map.put(e.getKey(), weightingMap.getDouble("highways." + e.getKey(), e.getValue()));
+            map.put(e.getKey(), pMap.getDouble(e.getKey(), e.getValue()));
         }
 
-        ConfigMap cMap = new ConfigMap();
-        cMap.put("highways", map);
+        Result result = new Result();
+        // create per request object
+        result.speedArray = getHighwaySpeedMap(map);
+        return result;
+    }
 
-        cloneDoubleAttribute(weightingMap, cMap, GenericWeighting.HEIGHT_LIMIT, 0d);
-        cloneDoubleAttribute(weightingMap, cMap, GenericWeighting.WEIGHT_LIMIT, 0d);
-        cloneDoubleAttribute(weightingMap, cMap, GenericWeighting.WIDTH_LIMIT, 0d);
+    public class Result {
+        private double[] speedArray;
 
-        return cMap;
-    }
+        public double getSpeed(EdgeIteratorState edgeState) {
```

Comment:
> check the speed encoder of the edge as well

What do you mean here? There is no speed for DataFlagEncoder. Only a max speed value.